### ISSUE TITLE
[Fix] SecretBox has wrong message authentication code (MAC) with 2.0.0

### ISSUE THREAD

**Issue by FrancoisJ** [dint-dev/cryptography#49]
Hi, 

I use chacha20 with previous version. My flutter app juste decrypt data, data is encrypted by [external cipher.](pycryptodome.org) When I migrated to new version I have this error `SecretBox has wrong message authentication code (MAC)`.

When I change in my code this line of code
```
UIntList data = ...
final box = SecretBox.fromConcatenation(data, nonceLength: 12, macLength: 16);
```
By this one
```
  final box = SecretBox(
      data.sublist(12, data.length-16),
      nonce: data.sublist(0, 12),
      mac: Mac(data.sublist(data.length-16))
  );
```

It works great again.

I found this line in `SecretBox.fromConcatenate` source code:
https://github.com/dint-dev/cryptography/blob/2c28aec5244d863d3c05896933cc31f0ac02099c/cryptography/lib/src/cryptography/secret_box.dart#L153

It seem that ciphertext take also the nonce. Is it correct ? Or we should remove nonce from ciphertext, with `data.offsetInBytes + nonceLength`.

**Comment by AhmetMelihSerter** [comment]
#51

**Comment by terrier989** [comment]
Fixed in 2.0.2.
